### PR TITLE
Migrate to kubermatic.k8c.io

### PR DESCRIPTION
**Pull request by xrstf**
### What this PR does / why we need it
This PR updates the API groups used for e2e tests to the new groups that are soon going to be used by KKP.

Currently, before https://github.com/kubermatic/kubermatic/pull/8783 is merged, this PR will always fail.

### Which issue(s) this PR fixes
This is part of kubermatic/kubermatic#6471

### Release note
```release-note
NONE
```

### CODE REVIEW

**Comment by maciaszczykm** [comment]
Updated names ❤️

File: hack/e2e/fixtures/seed.yaml [hunk 1]
```diff
@@ -116,23 +117,24 @@ spec:
...
-          dns_servers:
+          authURL: https://api.cbk.cloud.syseleven.net:5000/v3
+          availabilityZone: dbl1
+          dnsServers:
           - 37.123.105.116
           - 37.123.105.117
-          enforce_floating_ip: true
-          ignore_volume_az: false
+          enforceFloatingIP: true
+          ignoreVolumeAZ: false
```
Comment (maciaszczykm) [inline hack/e2e/fixtures/seed.yaml:132]: I can see these values are in https://github.com/kubermatic/dashboard/blob/master/cypress/fixtures/datacenters.json too.

File: hack/e2e/fixtures/seed.yaml [hunk 2]
```diff
@@ -84,7 +85,7 @@ spec:
       spec:
         gcp:
           region: europe-west3
-          zone_suffixes:
+          zoneSuffixes:
```
Comment (maciaszczykm) [inline hack/e2e/fixtures/seed.yaml:88]: It's also part of https://github.com/kubermatic/dashboard/blob/master/src/app/shared/entity/datacenter.ts.

**Comment by maciaszczykm** [comment]
Not all sources and fixtures are adjusted at the moment. The full tests can fail due to some provider issues but they also provide info what is not working, so let's not ignore them completely.

File: hack/e2e/fixtures/seed.yaml (hunk 2 above)
Comment (xrstf) [reply hack/e2e/fixtures/seed.yaml:88]: What are the files like datacenter.ts actually based on? The REST API in KKP still has the exact same types. Unless you use the CRDs 1:1 in the dashboard, I am not sure that the entity files need to be changed here.

Comment (maciaszczykm) [reply hack/e2e/fixtures/seed.yaml:88]: Ah, okay. So it's not updated in the API. Then ignore this comment.

Comment (xrstf) [reply hack/e2e/fixtures/seed.yaml:88]: Yeah we only changed the CRDs but kept the API _mostly_ the same. There are a few changes like https://github.com/kubermatic/dashboard/pull/4131/commits/dab8d7823e55d87326e3efa63908b758c7d3fad5 (and this one needs a bit of you guys' love, cause I don't know how to properly handle a string[] for the Presets). ^^

**Comment by floreks** [comment]
Someone finally did it and got rid of all those snake_case relics :heart: Thank you :slightly_smiling_face:

**Comment by xrstf** [comment]
Thank @kron4eg, he spearheaded the cleanup for our tags :)

**Comment by xrstf** [comment]
@maciaszczykm  /approve instead of /approved pls ^^

**Pull request merged**